include/Context.hpp

```diff
@@ -0,0 +1,21 @@
+#ifndef CONTEXT_HPP
+#define CONTEXT_HPP
+
+#include <Arduino.h>
+#include "EthernetDriver.hpp"
+
+class Context {
```

Comment:
Is Context supposed to be using a default constructor? Seems like you should make this explicit to better establish the purpose of the class. Also why exactly are the member variables protected?

Reply:
Made default constructor call explicit, changed member variables to private.

Reply:
The class isn't a singleton if that's what you were aiming for, gotta have static checks to verify only one object unless you are going for a naive implementation